modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseModelListenerCheck.java

```diff
@@ -124,7 +160,21 @@ private String _formatSuper(String javaMethodContent) {
 		return StringUtil.replace(javaMethodContent, methodCall, newMethodCall);
 	}
 
-	private static final Pattern _superPattern = Pattern.compile(
-		"super.\\s*\\w+\\(\\s*.+\\)");
+	private String _getExtendedClassType(String content) throws Exception {
+		Matcher matcher = _extendedClassTypePattern.matcher(content);
+
+		if (!matcher.find()) {
+			throw new Exception("Cannot find extended class type");
+		}
+
+		return matcher.group(1);
+	}
+
+	private static final Pattern _extendedClassTypePattern = Pattern.compile(
+		"extends\\s*\\w+\\<(.+?)\\>");
```

Comment:
extends\\s* <---- extends keyword + spacing to account for
\\w+ <---- it technically doesnt matter what class is getting extended, since its explicitly checked for in the code, using w+ as part of regex allows this to be easily migrated if we want to reuse
\\<(.+?)\\> <--- Match the type of the extended class if it has one. This will throw an exception if the extended class has no type

Reply:
Hi @kyle-miho, since doesn't compile without any spacing, should be `extends\s+`
Also, using `.+` it's too generic, and in general we should avoid types of generalizations. It opens up gaps for edge cases that we don't see. 
And there is a reason to make the quantifier lazy? Using `+?`,  "Makes the preceding quantifier lazy, causing it to match as few characters as possible. By default, quantifiers are greedy, and will match as many characters as possible." If hasn't we could remove it?

In the end, Could not be: `"extends\\s+\\w+\\<(\\w+)\\>"`?